simple_certmanager/models.py

```diff
@@ -52,64 +59,52 @@ def __str__(self):
         return self.label or gettext("(missing label)")
 
     @property
-    def _certificate(self):
-        if not self._certificate_obj:
+    def _certificate(self) -> CryptographyCertificate:
+        if self._certificate_obj is None:
             with self.public_certificate.open(mode="rb") as certificate_f:
-                self._certificate_obj = crypto.load_certificate(
-                    crypto.FILETYPE_PEM, certificate_f.read()
-                )
+                self._certificate_obj = load_pem_x509_certificate(certificate_f.read())
         return self._certificate_obj
 
     @property
     def _private_key(self):
-        if not self._private_key_obj:
+        if self._private_key_obj is None:
             with self.private_key.open(mode="rb") as key_f:
-                self._private_key_obj = crypto.load_privatekey(
-                    crypto.FILETYPE_PEM, key_f.read()
-                )
+                self._private_key_obj = load_pem_x509_private_key(key_f.read())
         return self._private_key_obj
 
     @property
     def expiry_date(self) -> datetime:
-        expiry = self._certificate.get_notAfter()
-        return datetime.strptime(expiry.decode("utf-8"), "%Y%m%d%H%M%SZ")
+        # TODO: should probably be stored in a DB column after saving the file so
+        # we can query on it to report (nearly) expired certificates.
+        return self._certificate.not_valid_after_utc
 
     @property
     def issuer(self) -> str:
-        issuer_x509name = self._certificate.get_issuer()
-        return pretty_print_certificate_components(issuer_x509name)
+        return pretty_print_certificate_components(self._certificate.issuer)
 
     @property
     def subject(self) -> str:
-        subject_x509name = self._certificate.get_subject()
-        return pretty_print_certificate_components(subject_x509name)
+        return pretty_print_certificate_components(self._certificate.subject)
 
     @property
     def serial_number(self) -> str:
-        x509sn: int = self._certificate.get_serial_number()
+        x509sn = self._certificate.serial_number
         sn: str = hex(x509sn)[2:].upper()
         bytes = (sn[i : i + 2] for i in range(0, len(sn), 2))
         return ":".join(bytes)
```

Comment:
the `str` representation of the serial number produces this but without colons, so I've left the implementation as is. Relying on the (undocumented) `str`/`repr` doesn't seem like a great idea.

Reply:
no, this was the cryptography variant in an interactive debugger. it threw me off when I thought it already returned string, but I was wrong.